app/Jobs/CreateCallPowerPostInRogue.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Chompy\Jobs;
+
+use Exception;
+use Chompy\Services\Rogue;
+use Illuminate\Http\Request;
+use Illuminate\Bus\Queueable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+
+
+class CreateCallPowerPostInRogue implements ShouldQueue
+{
+	use Dispatchable, InteractsWithQueue, Queueable;
+
+	/**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+	public function __construct()
+	{
+	}
+
+    /**
+     * Execute the job to create a CallPower post in Rogue.
+     *
+     * @return array
+     */
+    public function handle(Rogue $rogue, Request $request)
+    {
+		// Using the mobile number, get or create a northstar_id.
+    	$user = $this->getOrCreateUser($request['mobile']);
+
+        // Using the callpower_campaign_id, get the action_id from Rogue.
+    	$action = $rogue->getActionFromCallPowerCampaignId($request['callpower_campaign_id']);
+
+        // Check if the post exists in Rogue. If not, create the post.
+    	$existingPost = $rogue->getPost([
+    		'action_id' => $action['data'][0]['id'],
+    		'northstar_id' => $user->id,
+    	]);
+
+    	if (! $existingPost['data']) {
+    		info('creating post in rogue for northstar user: ' . $user->id);
+
+    		$details = $this->extractDetails($request);
+
+        	// Determine source details.
+    		$post = $rogue->createPost([
+    			'northstar_id' => $user->id,
+    			'action_id' => $action['data'][0]['id'],
+    			'type' => 'phone-call',
+    			'status' => $request['status'] === 'completed' ? 'accepted' : 'incomplete',
+    			'quantity' => 1,
+    			'source' => 'CallPower',
+    			'details' => $details,
+    			'dont_send_to_blink' = true,
```

Comment:
We definitely want this for historical records but not sure what we want to do for go-forward. Going to touch base with Jen when she is back from vacation!